src/embed.ts

```diff
@@ -162,14 +162,14 @@ export abstract class Embed {
      * @param {models.ILoadConfiguration} config
      * @returns {Promise<void>}
      */
-    load(config: models.ILoadConfiguration, embedConfig?: IEmbedConfiguration): Promise<void> {
+    load(config: models.ILoadConfiguration): Promise<void> {
         const errors = models.validateLoad(config);
         if(errors) {
             throw errors;
         }
         
         let loadPath = '/report/load';
-        if(embedConfig && embedConfig.type === 'dashboard') {
+        if(config && (<IEmbedConfiguration>config).type === 'dashboard') {
```

Comment:
**How will the overridden 'load' methods find out the load path info?**
I imagined these two ways:
1. Save on internal config
2. Save as loadPath class member.
It could also be supplied as optional argument but that doesn't solve the problem and still allows the type of the thing embedded to be changed independently of the object type.  I think the main point is that the type and the load path _should_ be coupled together, but as it is written now it technically works but is brittle design.

**wouldn't we have a lot of duplicated code in 3 load methods?**
No, we should try to avoid duplication.  The important part is the call to hpm which should still be in the abstract base embed class.

> All we want to change is the load path and I think the ideal place to differentiate is still in embed.
There is an existing programming model using objects and separate files for report and dashboard to express their differences.  It's better to keep consistency and have conceptual integrity across the library than to add things with conflicting philosophies even if you think they are better.  My personal preference would have been using the service for everything and using a more generic functional approach where there would be no dashboard or report classes only methods to invoke which send messages to an iframe but that is not what was agreed upon.  A lot of developers still like OOP.

> Technically you can't really prevent the user from doing crazy things such as embed a dashboard in a report iFrame
I think you're confusing two different ideas.  Yes, they can always do what they want with the iframe, but our SDK should not allow itself to be put in a bad state or allow the user to perform unsafe operations.  I think we should throw an exception in the embed code to protect the developers and then step back and re-design to better handle dynamically switching from one embed type to the other.  I think it's too much work to do now and I don't see the use case for it .  Josh might have specific examples, but I think embedding is almost always going to be based on the context of the host application. (E.g. on this page it makes sense embed a dashboard, but this page it makes sense to embed a report.  If you were two swap the report with the dashboard it would not feel correct in context of host application.)